File: module2-arrays/readability/readability.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
  // Get input from user
  string input = get_string("Text: ");

  // Initialize variables
  int letters = 0;
  int words = 1;
  int sentences = 0;

  // Loop through input
  for (int i = 0, n = strlen(input); i < n; i++)
  {
    // If character is a letter, increment letters
    if (isalpha(input[i]))
    {
      letters++;
    }

    // If character is a space, increment words
    if (isspace(input[i]))
    {
      words++;
    }

    // If character is a period, increment sentences
    if (input[i] == '.' || input[i] == '?' || input[i] == '!')
    {
      sentences++;
    }
  }

  // Calculate readability score
  float score = 0.0588 * (100.0 * letters / words) - 0.296 * (100.0 * sentences / words) - 15.8;

  // Print score
  if (score < 1)
  {
    printf("Before Grade 1\n");
  }
  else if (score >= 16)
  {
    printf("Grade 16+\n");
  }
  else
  {
    printf("Grade %.0f\n", score);
  }
}

File: pset6/readability/readability.py
from curses.ascii import isalpha, isspace


def main():
  text = get_string("Text: ")
  [letters, words, sentences] = resolve_text(text)
  score = round(0.0588 * (100.0 * letters / words) - 0.296 * \
      (100.0 * sentences / words) - 15.8)
  print_score(score)


def get_string(arg: str = ""):
  return input(arg)


def resolve_text(arg: str):
  letters = 0
  words = 1
  sentences = 0

  for i in arg:
    if isalpha(i):
      letters += 1
    if isspace(i):
      words += 1
    if i == '.' or i == '?' or i == '!':
      sentences += 1

  return [letters, words, sentences]


def print_score(score: int):
  if score < 1:
    print("Before Grade 1")
  elif score >= 16:
    print("Grade 16+")
  else:
    print(f"Grade {score}")


main()
